### PR TITLE
Add route to list entries with annotations

### LINKED ISSUE #3355: List all articles with annotations
It would be good to be able to list all articles with annotations (e.g., from the sidebar, alongside tags), maybe showing the title of the article, and a list of annotations with links to where in the article they were made.

It would also be good to subfilter that list through a full-text search box and tags.

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Documentation | no
| Translation   | yes
| CHANGELOG.md  | no
| License       | MIT

Fixes #3355

Todo:
- [x] tests

Capture d’écran 2020-04-20 à 19 03 35

### CODE REVIEW

**Comment by tcitworld** [comment]
I feel it shouldn't be a navigation item on the right but a filter option.

**Comment by nicosomb** [comment]
@wallabag/core what's your opinion about @tcitworld suggestion?

File: tests/Wallabag/CoreBundle/Controller/EntryControllerTest.php [hunk 1]
```diff
@@ -399,6 +399,16 @@ public function testStarred()
...
     }
 
+    public function testWithAnnotations()
+    {
+        $this->logInAs('admin');
+        $client = $this->getClient();
+
+        $client->request('GET', '/with_annotations/list');
+
+        $this->assertSame(200, $client->getResponse()->getStatusCode());
```
Comment (Kdecherf) [inline tests/Wallabag/CoreBundle/Controller/EntryControllerTest.php:409]: The test should ensure that the retrieved entries have actual annotations.

File: app/config/security.yml [hunk 2]
```diff
@@ -69,11 +69,11 @@ security:
         - { path: ^/logout, roles: [IS_AUTHENTICATED_ANONYMOUSLY, IS_AUTHENTICATED_2FA_IN_PROGRESS] }
         - { path: ^/register, role: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/resetting, role: IS_AUTHENTICATED_ANONYMOUSLY }
-        - { path: /(unread|starred|archive|all).xml$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: /(unread|starred|archive|with_annotations|all).xml$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
```
Comment (Kdecherf) [inline app/config/security.yml:72]: We should avoid `_` in the path and prefer things like `with-annotations`

**Comment by nicosomb** [comment]
Will fix. What do you think about @tcitworld ´s suggestion ?

**Comment by Kdecherf** [comment]
> Will fix. What do you think about @tcitworld ´s suggestion ?

Both would be ok imo. _Unread_, _starred_ and _archived_ are both navigation items and filter options so it could (_should?_) be the same for the annotations.

**Comment by nicosomb** [comment]
> Both would be ok imo. Unread, starred and archived are both navigation items and filter options so it could (should?) be the same for the annotations

So a checkbox in the filter menu?

**Comment by robertodijo** [comment]
Thanks dev. Hope this feature will be available soon 👏👏

File: src/Wallabag/CoreBundle/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -243,6 +245,7 @@ entry:
         archived_label: Archived
         starred_label: Starred
         unread_label: Unread
+        annotated_label: Annotated
```
Comment (Kdecherf) [inline src/Wallabag/CoreBundle/Resources/translations/messages.en.yml:248]: French translation for this entity is missing

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entries.html.twig [hunk 4]
```diff
@@ -130,6 +130,11 @@
                     {{ form_label(form.isUnread) }}
                 </div>
 
+                <div class="input-field col s6 with-checkbox">
+                    {{ form_widget(form.isAnnotated) }}
+                    {{ form_label(form.isAnnotated) }}
+                </div>
+
```
Comment (Kdecherf) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entries.html.twig:137]: I think we'll have issues with this label, we should put it on a new line alone.

2021-08-07-104020_233x128

**Review by Kdecherf — DISMISSED** [review]

**Comment by StanczakDominik** [comment]
Gentle ping for @Kdecherf, it'd be great if you could find some time for another review of this, as this PR would exactly solve my current issues with accessing notes from articles!

**Comment by j0k3r** [comment]
Force pushed to rebase against the 2.5.0 branch.

**Comment by nicosomb** [comment]
@Kdecherf could you review this PR please?

**Review by Kdecherf — APPROVED** [review]
Looks good

**Pull request merged**